state/storage.go

```diff
@@ -797,7 +806,7 @@ func readStorageConstraints(st *State, key string) (map[string]StorageConstraint
 	var doc storageConstraintsDoc
 	err := coll.FindId(key).One(&doc)
 	if err == mgo.ErrNotFound {
-		return nil, nil
+		return nil, errors.NotFoundf("storage constraints for %q", key)
```

Comment:
We should always have a constraints document, and should consider a missing document to be an error. For now, at the existing call-sites of this function I'm trapping the not-found error and converting to nil; just to conservative, because I don't want to break anything.